src/system.h

```diff
@@ -783,19 +783,36 @@ extern void SyAbortBags(const Char * msg) NORETURN;
 /****************************************************************************
 **
 *F * * * * * * * * * * * * * loading of modules * * * * * * * * * * * * * * *
+**
+**  Define the strings which define different module types.
+**
+**  These are built up as strings of the form: "T_KernelModuleVersion"
+**  Where:
+**   - T is one of the constants: 'B' - Builtin, 'S' - Static, 'D' - Dynamic
+**   - KernelVersion is a string
+**
+**  Note that KernelModuleVersion is NOT necessarily the same as the GAP
+**  version (4.4.2 for example), because we only want to change this string
+**  when we break kernel module compatibility, which we might not do in
+**  minor GAP releases.
+**
+**  The reason for putting the version in these strings is it provides a
+**  a simple and backwards-compatable method of injecting the kernel version
+**  into packages with kernel extensions, to check on load.
+**
+**  GAP will refuse to load any kernel module if this string does not match
+**  exactly.
+**
+**  When a breaking change is made to the kernel, GAP_KERNEL_MODULE_VERSION
+**  should be incremented.
 */
 
-enum {
-    /** builtin module */
-    MODULE_BUILTIN = 1,
-
-    /** statically loaded compiled module */
-    MODULE_STATIC  = 2,
-
-    /** dynamically loaded compiled module */
-    MODULE_DYNAMIC = 3,
-};
+#define GAP_KERNEL_MODULE_VERSION "4.dev_1"
 
+#define VERSION_STRING GAP_KERNEL_MODULE_VERSION
+#define MODULE_BUILTIN "B_" VERSION_STRING
+#define MODULE_STATIC "S_" VERSION_STRING
+#define MODULE_DYNAMIC "D_" VERSION_STRING
```

Comment:
Perhaps retain the three comments for these constants?

Also, best practice is to declare macros "as safe as possible" which in this case would suggest to put parens around them.